datanode/api/trading_data_v2.go

```diff
@@ -2723,15 +2727,22 @@ func (t *TradingDataServiceV2) EstimatePosition(ctx context.Context, req *v2.Est
 		var price num.Decimal
 		p, err := num.DecimalFromString(o.Price)
 		if err != nil {
-			return nil, errors.Wrap(ErrInvalidOrderPrice, err.Error())
+			return nil, formatE(ErrInvalidOrderPrice, err)
+		}
+
+		if p.IsNegative() {
```

Comment:
Given the docstring we say its an unsigned integer, I think we also need an `if !p.IsInteger{}` check here too?